interfaces/Portalicious/src/app/components/health-widget/health-widget.component.ts

```diff
@@ -0,0 +1,24 @@
+import { JsonPipe } from '@angular/common';
+import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
+import {
+  injectQuery,
+  injectQueryClient,
+} from '@tanstack/angular-query-experimental';
+import { ApiEndpoints, ApiService } from '~/services/api.service';
+
+@Component({
+  selector: 'app-health-widget',
+  standalone: true,
+  imports: [JsonPipe],
+  templateUrl: './health-widget.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class HealthWidgetComponent {
+  apiService = inject(ApiService);
+  queryClient = injectQueryClient();
+
+  versionInfo = injectQuery(() => ({
+    queryKey: [ApiEndpoints.versionInfo],
```

Comment:
I'm not too sure what a good way of handling these query keys would be. Using the endpoint seems like a good way to ensure that it is unique, and reflects the right endpoint, but it also seems like repetition. Maybe there is a smarter way to handle this but haven't thought of an obvious one, and this seems good enough (for now at least). Open to ideas though.

Reply:
just as a thought for the future, will we have the need to move the ApiEndpoints enum outside of the ApiService?

Reply:
Yeah, we potentially can. I prefer to keep the number of files to a minimum, because it makes things simpler for me in my mind - but if I'm the minority then I'm easily persuaded to move this to a separate file.